src/Elastic.Transport.VirtualizedCluster/Rules/RuleBase.cs

```diff
@@ -9,23 +9,34 @@ namespace Elastic.Transport.VirtualizedCluster.Rules
 {
 	public interface IRule
 	{
+		/// <summary>The value or exception to return after the call succeeds</summary>
 		RuleOption<Exception, int> AfterSucceeds { get; set; }
+
+		/// <summary>This rule is constrain on the node with this port number</summary>
 		int? OnPort { get; set; }
 
-		/// <summary>
-		/// Either a hard exception or soft HTTP error code
-		/// </summary>
+		/// <summary> Either a hard exception or soft HTTP error code</summary>
 		RuleOption<Exception, int> Return { get; set; }
 
+		/// <summary>Set an explicit return content type for the API call</summary>
 		string ReturnContentType { get; set; }
 
+		/// <summary>Explicitly set the bytes returned by the API call, optional.</summary>
 		byte[] ReturnResponse { get; set; }
+
+		/// <summary>Whether this rule describes an API call succeeding or not</summary>
 		bool Succeeds { get; set; }
+
+		/// <summary>Simulate a long running call</summary>
 		TimeSpan? Takes { get; set; }
+
+		/// <summary> The number of times this rule stays valid after being called</summary>
 		RuleOption<TimesHelper.AllTimes, int> Times { get; set; }
 
+		/// <summary> The amount of times this rule has been executed</summary>
 		int Executed { get; }
```

Comment:
Not related to this PR, but should we name this ExecuteCount for clarity?

Reply:
Created #16 to track this comment so we wont forget 👍